src/containerapp/azext_containerapp/_utils.py

```diff
@@ -1714,3 +1726,161 @@ def format_location(location=None):
+    # Check if the image minor version is less than the latest minor version
+    if tag_prop["version"] < matching_version_info[0]["version"]:
+        result["oldRunImage"] = tag_prop["fullTag"]
+        if (tag_prop["version"].minor == matching_version_info[0]["version"].minor) and (tag_prop["version"].micro < matching_version_info[0]["version"].micro):
+            # Patchable
+            result["newRunImage"] = "mcr.microsoft.com/oryx/builder:" + matching_version_info[0]["fullTag"]
+            result["id"] = hashlib.md5(str(result["oldRunImage"] + result["targetContainerName"] + result["targetContainerAppName"] + result["targetResourceGroup"] + result["newRunImage"]).encode()).hexdigest()
+            result["reason"] = "New security patch released for your current run image."
+        else:
+            # Not patchable
+            result["newRunImage"] = "mcr.microsoft.com/oryx/builder:" + matching_version_info[0]["fullTag"]
+            result["id"] = None
+            result["reason"] = "The image is not patchable. Please check for major or minor version upgrade."
+    else:
+        result["oldRunImage"] = tag_prop["fullTag"]
+        result["reason"] = "The image is already up to date."
+    return result
+
+
+def get_current_mariner_tags() -> list(OryxMarinerRunImgTagProperty):
+    r = requests.get("https://mcr.microsoft.com/v2/oryx/builder/tags/list", timeout=30)
+    tags = r.json()
+    tag_list = {}
+    # only keep entries that contain keyword "mariner"
+    tags = [tag for tag in tags["tags"] if "mariner" in tag]
+    for tag in tags:  # pylint: disable=too-many-nested-blocks
+        tag_obj = parse_oryx_mariner_tag(tag)
+        if tag_obj:
+            major_minor_ver = str(tag_obj["version"].major) + "." + str(tag_obj["version"].minor)
+            support = tag_obj["support"]
+            framework = tag_obj["framework"]
+            mariner_ver = tag_obj["marinerVersion"]
+            if framework in tag_list:
+                if major_minor_ver in tag_list[framework].keys():
+                    if support in tag_list[framework][major_minor_ver].keys():
+                        if mariner_ver in tag_list[framework][major_minor_ver][support].keys():
+                            tag_list[framework][major_minor_ver][support][mariner_ver].append(tag_obj)
+                            tag_list[framework][major_minor_ver][support][mariner_ver].sort(reverse=True, key=lambda x: x["version"])
+                        else:
+                            tag_list[framework][major_minor_ver][support][mariner_ver] = [tag_obj]
+                    else:
+                        tag_list[framework][major_minor_ver][support] = {mariner_ver: [tag_obj]}
+                else:
+                    tag_list[framework][major_minor_ver] = {support: {mariner_ver: [tag_obj]}}
+            else:
+                tag_list[framework] = {major_minor_ver: {support: {mariner_ver: [tag_obj]}}}
+    return tag_list
```

Comment:
It is better to flatten the loop.

Reply:
Updated: nested statements are flattened now.